### ISSUE TITLE
possible bug in io.imshow_collection for more than 6 images

### ISSUE THREAD

**Issue by TheShadow29** [scikit-image/scikit-image#2676]
- [x] Code example (attached inline)
- [x] Relevant images (attached inline)
- [x] Operating system and version : Ubuntu 16.04
- [x] Python version : 2.7.12
- [x] scikit-image version : 0.13.0

The issue is if I try to do imshow_collection on all the images from a list it shows some garbage values. 
```
%matplotlib
from skimage import io
io.use_plugin('matplotlib')
io.imshow_collection(io.imread_collection(file_list))
```
This outputs in the following image:

![screenshot from 2017-06-09 11-06-39](https://user-images.githubusercontent.com/10602903/26988318-ba9e0eaa-4d03-11e7-96b8-bdddabc49b6b.png)
But if I do the same thing by opening each image using `io.imread(img)` the image is shown properly. Below is the code for the same 
```
for f in file_list:
    plt.figure()
    io.imshow(io.imread(f))
```
I have also tried the following 
```
%matplotlib
from skimage import io
io.use_plugin('matplotlib')
io.imshow_collection(io.imread_collection(file_list[:6]))
```
Suprisingly this gives the correct result, but changing 6 to 7 again outputs a garbage image (I posted above). My suspicion is that there is an upper limit on the number of files that can be handled by `io.imshow_collection()`.

**Comment by soupault** [comment]
I don't see a bug here. It is rather a peculiarity of matplotlib plotting engine.

Here is how it look on my PC (latest versions of Python, skimage, matplotlib):
![figure_1](https://user-images.githubusercontent.com/1315589/27005671-aba15906-4e2b-11e7-8f0b-367db3e0cd51.png)

By the way, what is your `matplotlib` version?

**Comment by jni** [comment]
Well, at the minimum I would say that we should try to extract some approximate nrow and ncol factors from the number of images, and plot the images using those, rather than in one big row, which will be useless except for a tiny number of images.

**Comment by jni** [comment]
Oh LOL I just saw #2678. I'll go back into my hole now. =P

**Comment by tacaswell** [comment]
That is a fun corner case of fixed aspect ratio on `Axes` objects fighting back....

**Comment by stefanv** [comment]
This issue will be fixed by #2678.
